packages/@jsii/check-node/src/constants.ts

```diff
@@ -1,69 +1,161 @@
-export const enum SupportLevel {
+import { version } from 'process';
+import { Range, SemVer } from 'semver';
+
+/**
+ * The support information for a given node release range.
+ *
+ * @see https://nodejs.org/en/about/releases/
+ */
+export class NodeRelease {
   /**
-   * Soft-deprecated, removed as of releases made starting at DEADLINE. This
-   * will output a yellow warning to the console, until the DEADLINE has been
-   * reached, at which point it'll be a SCARY RED WARNING.
+   * How long before enf-of-life do we start warning customers? Expressed in
+   * milliseconds to make it easier to deal with JS dates.
    */
-  DEPRECATED = 'deprecated',
+  private static readonly DEPRECATION_WINDOW_MS = 30 * 86_400_000;
```

Comment:
Can we make this magic number less magical (perhaps define `const ONE_DAY = 86_400_000;` somewhere)?